src/main/java/org/embulk/input/zendesk/utils/ZendeskUtils.java

```diff
@@ -0,0 +1,167 @@
+package org.embulk.input.zendesk.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Charsets;
+import org.embulk.input.zendesk.models.Target;
+
+import org.embulk.spi.Column;
+import org.embulk.spi.ColumnVisitor;
+
+import org.embulk.spi.Exec;
+import org.embulk.spi.PageBuilder;
+import org.embulk.spi.Schema;
+import org.embulk.spi.json.JsonParser;
+import org.embulk.spi.time.Timestamp;
+import org.slf4j.Logger;
+
+import java.util.Base64;
+
+public class ZendeskUtils
+{
+    private static final Logger logger = Exec.getLogger(ZendeskUtils.class);
+
+    private ZendeskUtils()
+    {}
+
+    public static boolean isSupportIncremental(final Target target)
+    {
+        return !Target.TICKET_FORMS.equals(target)
+                && !Target.TICKET_FIELDS.equals(target);
+    }
+
+    public static String convertBase64(final String text)
+    {
+        return Base64.getEncoder().encodeToString(text.getBytes(Charsets.UTF_8));
+    }
+
+    public static int numberToSplitWithHintingInTask(int count)
+    {
+        return count % ZendeskConstants.Misc.RECORDS_SIZE_PER_PAGE == 0
+                ? count / ZendeskConstants.Misc.RECORDS_SIZE_PER_PAGE
+                : (count / ZendeskConstants.Misc.RECORDS_SIZE_PER_PAGE) + 1;
+    }
+
+    public static synchronized void addRecord(JsonNode record, Schema schema, PageBuilder pageBuilder)
+    {
+        schema.visitColumns(new ColumnVisitor() {
```

Comment:
Would suggest:
* Add 1 more common method:
```
private <T> void setColumn(Column column, JsonNode data, T value, Function<T, Void> setter) {
                if (isNull(data)) {
                    pageBuilder.setNull(column);
                    return;
                }
                setter.apply(value);
            }
```
* Update other columns use that utility method e.g.
```
@Override
            public void doubleColumn(Column column)
            {
                JsonNode data = record.get(column.getName());

                setColumn(column, data, data.asDouble(), (value) -> {
                    pageBuilder.setDouble(column, value);
                    return null;
                });
            }
```

Reply:
data.asDouble() will throw NPE when data is null

Reply:
my bad since I recognized that one but copied the old snippet to here ..., it should be:
```
@Override
            public void doubleColumn(Column column)
            {
                JsonNode data = record.get(column.getName());

                setColumn(column, data, (value) -> {
                    pageBuilder.setDouble(column, data.asDouble());
                    return null;
                });
            }
```
Function `setDouble` will only be explicitly called which happen only once `null` is already checked.